### PR TITLE
Add go infra tools

### PR DESCRIPTION
**Pull request by RamLavi**
Adding go infra tools that will assist the developers of these repo.

### CODE REVIEW

File: hack/go-mod-version.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash -e
+grep ^go go.mod |awk '{print $2}'
```
Comment (orelmisan) [inline hack/go-mod-version.sh:2]: nit:
```suggestion
grep ^go go.mod | awk '{print $2}'
```

File: Makefile [hunk 2]
```diff
@@ -71,3 +71,7 @@ vendor:
 	           -e GOFLAGS=-mod=mod \
 	           $(GO_IMAGE_NAME):$(GO_IMAGE_TAG) go mod tidy -compat=$(GO_MOD_VERSION) && go mod vendor
 .PHONY: vendor
+
+check-uncommitted:
```
Comment (orelmisan) [inline Makefile:75]: Is this target intentionally manual?
It seems fit to be dependency of the `check` target.

Comment (RamLavi) [reply Makefile:75]: Nope, it's my bad..

File: Makefile [hunk 3]
```diff
@@ -51,3 +51,11 @@ fmt:
 	           $(GO_IMAGE_NAME):$(GO_IMAGE_TAG) gofmt -w .
 .PHONY: fmt
 
+vet:
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply Makefile:59]: ah thanks, will remove this then

File: Makefile [hunk 4]
```diff
@@ -43,3 +43,11 @@ lint:
 	            $(LINTER_IMAGE_NAME):$(LINTER_IMAGE_TAG) golangci-lint run --timeout 3m ./cmd/...
 .PHONY: lint
 
+fmt:
+	$(CRI_BIN) run --rm \
+	           --volume `pwd`:$(CURDIR):Z \
+	           --workdir $(CURDIR) \
+	           -e GOFLAGS=-mod=mod \
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply Makefile:55]: ah it was copied by mistake, thanks!

File: Makefile [hunk 5]
```diff
@@ -43,3 +43,11 @@ lint:
...
+	           $(GO_IMAGE_NAME):$(GO_IMAGE_TAG) gofmt -w .
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply Makefile:56]: I checked it by adding some spaces on cmd/main.go then saw that `make fmt` fixes them..

File: Makefile [hunk 6]
```diff
@@ -59,3 +60,10 @@ vet:
 	           $(GO_IMAGE_NAME):$(GO_IMAGE_TAG) go vet ./...
 .PHONY: vet
 
+vendor:
+	$(CRI_BIN) run --rm \
+	           --volume `pwd`:$(CURDIR):Z \
+	           --workdir $(CURDIR) \
+	           -e GOFLAGS=-mod=mod \
```
[the remark this answers is not in the document]
Comment (RamLavi) [reply Makefile]: it does not. removed

File: hack/go-mod-version.sh (hunk 1 above)
Comment (RamLavi) [reply hack/go-mod-version.sh]: fixed

**Review by orelmisan — APPROVED** [review]

**Pull request merged**